go/obscuronode/enclave/sql/edbattestation.go

```diff
@@ -0,0 +1,204 @@
+
+// ERA (Edgeless remote attestation) is a simple protocol for verifying edgeless tools. It's basically just a small json schema
+// that they use as a standard data blob to encrypt into their attestation reports, includes signerID, security version etc.
+
+// The only changes from https://github.com/edgelesssys/era/blob/master/era/era.go is the use of enclave.VerifyRemoteReport
+// in place of eclient.VerifyRemoteReport and a few tweaks to get our lint to pass.
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"crypto/tls"
+	"encoding/binary"
+	"encoding/hex"
+	"encoding/json"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+
+	"github.com/edgelesssys/ego/attestation"
+	"github.com/edgelesssys/ego/attestation/tcbstatus"
+	"github.com/edgelesssys/ego/enclave"
+	"github.com/tidwall/gjson"
+)
+
+type certQuoteResp struct {
+	Cert  string
+	Quote []byte
+}
+
+// ErrEmptyQuote defines an error type when no quote was received. This likely occurs when the host is running in OE Simulation mode.
+var ErrEmptyQuote = errors.New("no quote received")
+
+// GetCertificate gets the TLS certificate from the server in PEM format. It performs remote attestation
+// to verify the certificate. A config file must be provided that contains the attestation metadata.
+func GetCertificate(host, configFilename string) ([]*pem.Block, tcbstatus.Status, error) {
+	config, err := ioutil.ReadFile(configFilename)
+	if err != nil {
+		return nil, tcbstatus.Unknown, err
+	}
+	return getCertificate(host, config, enclave.VerifyRemoteReport)
+}
+
+// InsecureGetCertificate gets the TLS certificate from the server in PEM format, but does not perform remote attestation.
+func InsecureGetCertificate(host string) ([]*pem.Block, error) {
+	certs, _, err := getCertificate(host, nil, nil)
+	return certs, err
+}
+
+type verifyFunc func([]byte) (attestation.Report, error)
+
+func getCertificate(host string, config []byte, verifyRemoteReport verifyFunc) ([]*pem.Block, tcbstatus.Status, error) {
+	// we skip verify because we have the security of the attestation
+	cert, quote, err := httpGetCertQuote(&tls.Config{InsecureSkipVerify: true}, host, "quote") //nolint:gosec
+	if err != nil {
+		return nil, tcbstatus.Unknown, err
+	}
+
+	var certs []*pem.Block
+	block, rest := pem.Decode([]byte(cert))
+	if block == nil {
+		return nil, tcbstatus.Unknown, errors.New("could not parse certificate")
+	}
+	certs = append(certs, block)
+
+	// If we get more than one certificate, append it to the slice
+	for len(rest) > 0 {
+		block, rest = pem.Decode(rest)
+		if block == nil {
+			return nil, tcbstatus.Unknown, errors.New("could not parse certificate chain")
+		}
+		certs = append(certs, block)
+	}
+
+	if verifyRemoteReport == nil {
+		return certs, tcbstatus.Unknown, nil
+	}
+
+	if len(quote) == 0 {
+		return nil, tcbstatus.Unknown, ErrEmptyQuote
+	}
+
+	report, verifyErr := verifyRemoteReport(quote)
+	if verifyErr != nil && !errors.Is(verifyErr, attestation.ErrTCBLevelInvalid) {
```

Comment:
@BedrockSquirrel Can you add a comment here explaining why an `ErrTCBLevelInvalid` error is acceptable?